sdk/monitor/Microsoft.OpenTelemetry.Exporter.AzureMonitor/src/AzMonListExtensions.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.OpenTelemetry.Exporter.AzureMonitor
+{
+    internal static class AzMonListExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static object GetTagValue(this AzMonList tagObjects, string tagName)
+        {
+            for (int i = 0; i < tagObjects.Length; i++)
+            {
+                if (tagObjects[i].Key == tagName)
+                {
+                    return tagObjects[i].Value;
+                }
+            }
+
+            return null;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static object[] GetTagValues(this AzMonList tagObjects, params string[] tagNames)
+        {
+            int? length = tagNames?.Count();
```

Comment:
Curious - is it possible to have `tagNames == null`?

Reply:
It's our internal implementation. `tagNames` will always have value, could remove this check.